cmip6_downscaling/data/cmip.py

```diff
@@ -158,3 +161,74 @@ def cmip():
     print(f'done with cmip but these keys failed: {failed}')
 
     return model_dict, data
+
+
+def load_cmip_dictionary(
+    activity_ids=["CMIP"],
+    experiment_ids=["historical"],  # , "ssp126", "ssp245",  "ssp585"
+    member_ids=["r1i1p1f1"],
+    source_ids=["MIROC6"],  # BCC-CSM2-MR"]
+    table_ids=["day"],
+    grid_labels=["gn"],
+    variable_ids=["tasmax"],
+    return_type='zarr',
+):
+    """Loads CMIP6 GCM dataset dictionary based on input criteria.
+
+    Parameters
+    ----------
+    activity_ids : list, optional
+        [activity_ids in CMIP6 catalog], by default ["CMIP", "ScenarioMIP"],
+    experiment_ids : list, optional
+        [experiment_ids in CMIP6 catalog], by default ["historical", "ssp370"],  ex:#  "ssp126", "ssp245",  "ssp585"
+    member_ids : list, optional
+        [member_ids in CMIP6 catalog], by default ["r1i1p1f1"]
+    source_ids : list, optional
+        [source_ids in CMIP6 catalog], by default ["MIROC6"]
+    table_ids : list, optional
+        [table_ids in CMIP6 catalog], by default ["day"]
+    grid_labels : list, optional
+        [grid_labels in CMIP6 catalog], by default ["gn"]
+    variable_ids : list, optional
+        [variable_ids in CMIP6 catalog], by default ['tasmax']
+
+    Returns
+    -------
+    [dictionary]
+        [dictionary containing available xarray datasets]
+    """
+    col_url = "https://cmip6downscaling.blob.core.windows.net/cmip6/pangeo-cmip6.json"
+    print("intake")
+    store = (
+        intake.open_esm_datastore(col_url)
+        .search(
+            activity_id=activity_ids,
+            experiment_id=experiment_ids,
+            member_id=member_ids,
+            source_id=source_ids,
+            table_id=table_ids,
+            grid_label=grid_labels,
+            variable_id=variable_ids,
+        )
+        .df['zstore'][0]
+    )
+    if return_type == 'zarr':
+        return zarr.open_consolidated(store, mode='r')
+    elif return_type == 'xr':
+        return xr.open_zarr(store, consolidated=True)
```

Comment:
hmm neither of these options work for me when i tried to run this with the default argument in an ipython notebook. the zarr option fails because it's missing options like account name, and the xr option fails with `.zmetadata` not found. can you specify how you're using this function?